### PR TITLE
Add user event

### PR DESCRIPTION
**Pull request by quanvm**

### CODE REVIEW

File: EventInterface.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace go1\util\publishing\event;
+
+interface EventInterface
```
Comment (andytruong) [inline EventInterface.php]: Please provide description for each method. Thanks.

File: MQEventInterface.php [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?php
+
+namespace go1\util\publishing\event;
+
+interface MQEventInterface
+{
+    public function process(EventInterface $event);
```
Comment (andytruong) [inline MQEventInterface.php]: Event processes an event? It is not name.

Comment (quanvm) [reply MQEventInterface.php]: MQEventHandlerInterface ???

File: Event.php [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        $this->payload['embedded'][$key] = $value;
+    }
+
+    public function embedded()
+    {
+        /**
+         * @var EventPipelineInterface $pipeline
+         */
+        foreach ($this->pipelines as $pipeline) {
```
Comment (andytruong) [inline Event.php]: How can we add more pipe to $this->pipelines?

File: tests/user/UserEventTest.php [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    use UserMockTrait;
+    use PortalMockTrait;
+
+    private $mail       = 'abc@mail.com';
+    private $profileId  = 123;
+    private $instance   = 'qa.mygo1.com';
+
+    public function testFormat()
+    {
+        $this->createPortal($this->db, ['title' => $this->instance]);
```
Comment (andytruong) [inline tests/user/UserEventTest.php]: `instance` should be changed?

File: Event.php (hunk 3 above)
Comment (quanvm) [reply Event.php]: added setPipelines()

**Pull request merged**